charts/altinn-app/templates/deployment.yaml

```diff
@@ -0,0 +1,104 @@
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: {{ .Values.appName }}
+  labels:
+    {{- include ".Chart.Name.labels" . | nindent 4 }}
+spec:
+  {{- if not .Values.autoscaling.enabled }}
+  replicas: {{ .Values.replicaCount }}
+  {{- end }}
+  selector:
+    matchLabels:
+      {{- include ".Chart.Name.selectorLabels" . | nindent 6 }}
+  template:
+    metadata:
+      {{- if .Values.linkerd.enabled }}
+      annotations:
+        linkerd.io/inject: enabled
+        config.linkerd.io/skip-outbound-ports: 9092,10255,9093,443
+        azure.workload.identity/skip-containers: linkerd-proxy;linkerd-init
+      {{- end }}
+      labels:
+        {{- include ".Chart.Name.selectorLabels" . | nindent 8 }}
+        azure.workload.identity/use: "true"
+    spec:
+      serviceAccountName: {{ .Values.appName }}
+      containers:
+        - name: {{ .Values.appName }}
+          image: "{{ .Values.image.repository }}:{{ required "image.tag not set" .Values.image.tag }}"
+          imagePullPolicy: {{ .Values.image.pullPolicy }}
+          securityContext:
+            runAsUser: 1000
+            runAsGroup: 3000
+            allowPrivilegeEscalation: false
+          ports:
+            - containerPort: {{ .Values.image.containerPort }}
+          env:
+            {{- if not (hasKey .Values.environmentVariables .Values.environment) }}
+              {{ fail "the chosen environment does not exist" }}
+            {{- end }}
+            {{- $environmentVariables := index $.Values.environmentVariables $.Values.environment }}
+            {{- range $variable := $environmentVariables }}
+            - name: {{ $variable.name }}
+              {{- if $variable.value }}
+              value: {{ $variable.value | quote }}
+              {{- end }}
+              {{- if $variable.valueFrom }}
+              valueFrom:
+                {{- if $variable.valueFrom.secretKeyRef}}
+                secretKeyRef:
+                  name: {{ $variable.valueFrom.secretKeyRef.name }}
+                  key: {{ $variable.valueFrom.secretKeyRef.key }}
+                {{- end }}
+              {{- end }}
+            {{- end }}
+          readinessProbe:
+            httpGet:
+              path: /health
+              port: {{ .Values.service.internalPort }}
+
+            initialDelaySeconds: 30
+            failureThreshold: 3
+            periodSeconds: 3
+            timeoutSeconds: 1
+          livenessProbe:
+              httpGet:
+                path: /health
+                port: {{ .Values.service.internalPort }}
+              initialDelaySeconds: 3
+              failureThreshold: 3
+              periodSeconds: 10
+              timeoutSeconds: 2
```

Comment:
Should we make it possible to change the liveness/readiness probes?
Also should we add a default startup probe?